core/src/Cuda/Kokkos_Cuda_Instance.cpp

```diff
@@ -132,13 +132,11 @@ void cuda_device_synchronize(const std::string &name) {
       // annotate with __host__ silence a clang warning about using
       // cudaDeviceSynchronize in device code
       [] __host__() {  // TODO: correct device ID
-        KOKKOS_IMPL_CUDA_SAFE_CALL(cudaDeviceSynchronize());
-      });
 #else
       []() {  // TODO: correct device ID
+#endif
         KOKKOS_IMPL_CUDA_SAFE_CALL(cudaDeviceSynchronize());
       });
-#endif
```

Comment:
This isn't really changing the logic, and again may conflict with #5989. I'd suggest reverting this one.

Reply:
Ok, done.